frontend/src/resources/provider-connection.ts

```diff
@@ -17,25 +17,22 @@ export const ProviderConnectionDefinition: IResourceDefinition = {
 }
 
 export interface ProviderConnectionStringData {
-    awsAccessKeyID?: string
-    awsSecretAccessKeyID?: string
+    aws_access_key_id?: string
+    aws_secret_access_key?: string
 
     baseDomainResourceGroupName?: string
-    clientId?: string
-    clientSecret?: string
-    subscriptionId?: string
-    tenantId?: string
+    ['osServicePrincipal.json']?: string
```

Comment:
What is this [' '] notation for?


Other general comment is I would prefer to keep prefixes on variables to make it easier to know which provider they belong to.  Plus it seems to easy to have name collision with this change.

Reply:
Any keys that have a '.' in them need to be quoted.
Renaming the keys to match the names used by hive.